pkg/cmd/codespace/create.go

```diff
@@ -325,10 +341,14 @@ func (a *App) handleAdditionalPermissions(ctx context.Context, createParams *api
 	return codespace, nil
 }
 
-// showStatus polls the codespace for a list of post create states and their status. It will keep polling
-// until all states have finished. Once all states have finished, we poll once more to check if any new
+type codespaceStatusCheck struct{}
```

Comment:
since we're in the `codespace` package and this is unexported you could just call this `statusCheck` and `statusChecker` for the interface. its less to type and a bit cleaner IMO. same for `codespaceSSHClient`